cargo-shuttle/src/lib.rs

```diff
@@ -1656,11 +1658,26 @@ impl Shuttle {
         let client = self.client.as_ref().unwrap();
         let working_directory = self.ctx.working_directory();
 
-        let mut deployment_req: DeploymentRequest = DeploymentRequest {
+        let mut deployment_req = DeploymentRequest {
             no_test: args.no_test,
             ..Default::default()
         };
 
+        if self.beta {
+            let manifest_path = working_directory.join("Cargo.toml");
+            if !manifest_path.exists() {
+                bail!("Cargo manifest file not found: {}", manifest_path.display());
+            }
+            let metadata = async_cargo_metadata(manifest_path.as_path()).await?;
+            let packages = find_shuttle_packages(&metadata)?;
+            let package_name = packages
+                .first()
+                .expect("at least one shuttle crate in the workspace")
+                .name
+                .to_owned();
+            deployment_req.package_name = Some(package_name);
+        }
```

Comment:
Could we have a problem if there are more and the chosen name is not the one with the shuttle main function? Perhaps I'm not seeing the bigger picture here, and it doesn't matter. :thinking: